src/ansys/geometry/core/tools/prepare_tools.py

```diff
@@ -0,0 +1,83 @@
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Provides tools for preparing bodies."""
+
+from ansys.api.geometry.v0.bodies_pb2 import (
+    GetRequest,
+    GetType,
+)
+from ansys.api.geometry.v0.bodies_pb2_grpc import BodiesStub
+from ansys.api.geometry.v0.preparetools_pb2 import (
+    ShareTopologyRequest,
+)
+from ansys.api.geometry.v0.preparetools_pb2_grpc import PrepareToolsStub
+from beartype.typing import TYPE_CHECKING, List
+from google.protobuf.wrappers_pb2 import BoolValue, DoubleValue
+
+from ansys.geometry.core.connection import GrpcClient
+from ansys.geometry.core.typing import Real
+
+if TYPE_CHECKING:  # pragma: no cover
+    from ansys.geometry.core.designer.body import Body
+
+
+class PrepareTools:
+    """Prepare tools for PyAnsys Geometry."""
+
+    def __init__(self, grpc_client: GrpcClient):
+        """Initialize Prepare Tools class."""
+        self._grpc_client = grpc_client
+        self._prepare_stub = PrepareToolsStub(self._grpc_client.channel)
+        self._bodies_stub = BodiesStub(self._grpc_client.channel)
+
+    def share_topology(self, bodies: List["Body"], tol: Real = 0.0, p_instances: bool = False) -> bool:
+        """Apply ShareTopology to the chosen bodies.
+
+        Parameters
+        ----------
+        bodies : List[Body]
+            List of bodies to apply ShareTopology to.
+        tol : Real
+            The maximum distance between bodies.
+        p_instances : bool
+            Controls whether instances are preserved.
+
+        Returns
+        -------
+        bool
+            Indicates success/failure of ShareTopology operation.
+        """
+        if not bodies:
+            return False
+
+        tolerance_value = DoubleValue(value=float(tol))
+        preserve_instances_value = BoolValue(value=p_instances)
+        bodies_value = [
+            self._bodies_stub.Get(GetRequest(id=body.id, body_type=GetType.ORIGINAL)) for body in bodies
+        ]
+
```

Comment:
Why is this part needed? You already have access to the bodies.

Reply:
@RobPasMue I have access to the bodies, but (to the best of my knowledge) not to all the information a Body proto message should have. In particular, I don't think that the python bodies can give me a transform_to_master matrix, and if they can fill the can_suppress, is_deleted, or is_suppressed fields, I don't know how.